pep-9999.rst

```diff
@@ -514,6 +555,61 @@ No::
         forward_reference: 'OtherClass'
     class OtherClass: ...
 
+Version and Platform Checks
+---------------------------
+
+Use the tuple-based syntax for version checks. This is easier to read and write
+than the alternatives. Version checks should only check against the ``major``
+and ``minor`` parts of ``sys.version_info``.
+
+Yes::
+
+    if sys.version_info >= (3,):
+        # Python 3-specific type hints.
+    else:
+        # Python 2-specific type hints.
+
+    if sys.version_info >= (3, 5):
+        # Specific minor version features can be easily checked with tuples.
+
+    if sys.version_info < (3,):
+        # This is only necessary when a feature has no Python 3 equivalent.
+        # "<= (2, 7)" does not work because e.g. (2, 7, 1) > (2, 7).
+
+No::
+
+    if sys.version_info >= (3, 5, 1):
+        # Version checks should only check major and minor version numbers.
+
+    if sys.version_info[0] >= 3 and sys.version_info[1] >= 5:
+        # Just use a single tuple comparison: "sys.version_info >= (3, 5)"
+
+    if sys.version_info.major != 2:
+        # This is less clear than `sys.version_info >= (3,)`.
+
+Version checks should be kept as simple as possible, though more complex checks
+are occasionally necessary. For example, type hints for a feature that was added
+in Python 3.6 and backported to 2.7 may be guarded by ``if sys.version_info <
+(3,) or sys.version_info >= (3, 6)``.
+
+For platform checks, use simple equality comparisons between ``sys.platform``
+and the name of the platform as a string literal.
+
+Yes::
+
+    if sys.platform == 'win32':
+        # Windows-specific type hints.
+    else:
+        # Posix-specific type hints.
+
+No::
+
+    if sys.platform.startswith('linux'):
+        # Not necessary since Python 3.3.
+
+    if sys.platform in ['linux', 'cygwin', 'darwin']:
+        # Only '==' should be used in platform checks.
```

Comment:
I think "no" part can actually be moved to the first part. I don't think type checkers would support those constructs.